test/test_global_qbx.py

```diff
@@ -76,22 +93,23 @@ def iter_elements(discr):
             yield ElementInfo(
                 element_nr=element_nr,
                 discr_slice=slice(discr_nodes_idx,
-                   discr_nodes_idx + discr_group.nunit_nodes))
+                    discr_nodes_idx + discr_group.nunit_dofs))
 
-            discr_nodes_idx += discr_group.nunit_nodes
+            discr_nodes_idx += discr_group.nunit_dofs
 
 
 def run_source_refinement_test(ctx_factory, mesh, order,
         helmholtz_k=None, visualize=False):
```

Comment:
Not quite sure where to put this, but it might be a good idea to run the tests with `visualize=True` once to make sure that code didn't get stale.

Reply:
I tried to do that as I was adapting the tests, but I don't remember if I did it for all of them. It's possible some of these are already stale to begin with since they haven't been regularly tested.

Suggestion: A CI job that runs periodically to specifically check visualizations. It can be based on a non-interactive matplotlib backend. We can add an environment variable to control if the visualization gets run or not to make it easier to automate the testing.

Reply:
I agree with the thought, but I think it's out of scope for this PR. Resolving here, created an issue: https://github.com/inducer/pytential/issues/17